rfc/rfc-63/rfc-63.md

```diff
@@ -0,0 +1,370 @@
+file-level column stats less effective.
+
+### Use Case 3: File pruning support after partition evolution
+
+Partition evolution refers to the process of changing the partition columns for writing data to the storage. This
+requirement comes up when a user would like to reduce the number of physical partitions and improve the file sizing.
+
+Consider a case where event logs are stream from microservices and ingested into a raw event table. Each event log
+contains a `timestamp` and an associated organization ID (`org_id`). Most queries on the table are organization specific
+and fetch logs for a particular time range. A user may attempt to physically partition the data by both `org_id`
+and `date(timestamp)`. If there are 1K organization IDs and one year of data, such a physical partitioning scheme writes
+at least `365 days x 1K IDs = 365K` data files under 365K partitions. In most cases, the data can be highly skewed based
+on the organizations, with most organizations having less data and a handful of organizations having the majority of the
+data, so that there can be many small data files. In such a case, the user may want to evolve the partitioning by
+using `org_id` only without rewriting existing data, resulting in the physical layout of data like below
+
+| Physical partition path      | File Name            | Min of datestr | Max of datestr | Note                    |
+|------------------------------|----------------------|----------------|----------------|-------------------------|
+| org_id=1/datestr=2022-10-01/ | base_file_1.parquet  | `2022-10-01`   | `2022-10-01`   | Old partitioning scheme |
+| org_id=1/datestr=2022-10-02/ | base_file_2.parquet  | `2022-10-02`   | `2022-10-02`   |                         |
+| org_id=2/datestr=2022-10-01/ | base_file_3.parquet  | `2022-10-01`   | `2022-10-01`   |                         |
+| org_id=3/datestr=2022-10-01/ | base_file_4.parquet  | `2022-10-01`   | `2022-10-01`   |                         |
+| ...                          | ...                  | ...            | ...            | ...                     |
+| org_id=1/                    | base_file_10.parquet | `2022-10-10`   | `2022-10-11`   | New partitioning scheme |
+| org_id=2/                    | base_file_11.parquet | `2022-10-10`   | `2022-10-15`   |                         |
+| ...                          | ...                  | ...            | ...            | ...                     |
+
+As queries need to look for data for a particular time range, instead of relying on partition pruning, we can still use
+file pruning with file-level column stats. For the example above, even in the new partitioning scheme, without data
+being physically partitioned by the datestr, the data can still be co-located based the date, because of natural
+ingestion order or Hudi's clustering operation. In this case, we can effectively prune files based on the range of
```

Comment:
in new partitioning schema, if data is lately arrived at hudi, and the data goes to `base_file_11.parquet`, the min/max range would be very large, so the file pruning performance is still poor, how can we handle the situation? would only by clustering?